fuzz/fuzz_targets/same_result.rs

```diff
@@ -185,44 +185,3 @@ fn assert_val_eq(a: &Val, b: &Val) {
         _ => panic!("{:?} != {:?}", a, b),
     }
 }
-
-#[derive(Clone, Default, Debug)]
-struct WasmConfig;
-
-impl<'a> arbitrary::Arbitrary<'a> for WasmConfig {
-    fn arbitrary(_: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
-        Ok(WasmConfig)
-    }
-}
-
-impl wasm_smith::Config for WasmConfig {
-    fn max_memories(&self) -> usize {
-        10
-    }
-
-    fn max_memory_pages(&self) -> u32 {
-        // We want small memories that are quick to compare, but we also want to
-        // allow memories to grow so we can shake out any memory-growth-related
-        // bugs, so we choose `2` instead of `1`.
-        2
-    }
-
-    fn min_funcs(&self) -> usize {
-        // Always generate at least one function that we can hopefully use as an
-        // initialization function.
-        1
-    }
-
-    fn min_exports(&self) -> usize {
-        // Always at least one export, hopefully a function we can use as an
-        // initialization routine.
-        1
-    }
-
-    fn memory_offset_choices(&self) -> (u32, u32, u32) {
-        // Always use an offset immediate that is within the memory's minimum
-        // size. This should make trapping on loads/stores a little less
-        // frequent.
-        (1, 0, 0)
-    }
-}
```

Comment:
I don't think we want to stop providing these configuration settings. The way that `wasm-smith` is configured has indeed changed a little though.

Instead of having the fuzz target take a `wasm_smith::Module`, we will want it to take the raw data from the fuzzer, manually construct an `arbitrary::Unstructured` from that data, make an arbitrary `wasm_smith::Config` with that `Unstructured`, and then apply the above settings to that config before generating a `wasm_smith::Module`. Something like this:


```rust
fuzz_target!(data: &[u8]) {
    let mut u = Unstructured::new(data);

    let mut config = wasm_smith::Config::arbitrary(&mut u)?;
    config.min_funcs = 1;
    // etc...

    let module = wasm_smith::Module::new(config, &mut u)?;

    // and then continue with the fuzz target as before...

});
```

Reply:
thank you for the fast reply, I applied your suggestion :)